plugins/barcode-scanner/README.md

```diff
@@ -0,0 +1,98 @@
+
+Allows your mobile application to use the camera to scan QR codes, EAN-13 and other kinds of barcodes.
+
+## Install
+
+_This plugin requires a Rust version of at least **1.64**_
+
+There are three general methods of installation that we can recommend.
+
+1. Use crates.io and npm (easiest, and requires you to trust that our publishing pipeline worked)
+2. Pull sources directly from Github using git tags / revision hashes (most secure)
+3. Git submodule install this repo in your tauri project and then use file protocol to ingest the source (most secure, but inconvenient to use)
+
+Install the Core plugin by adding the following to your `Cargo.toml` file:
+
+`src-tauri/Cargo.toml`
+
+```toml
+[dependencies]
+tauri-plugin-barcode-scanner = "2.0.0-alpha"
+# alternatively with Git:
+tauri-plugin-barcode-scanner = { git = "https://github.com/tauri-apps/plugins-workspace", branch = "v2" }
+```
+
+You can install the JavaScript Guest bindings using your preferred JavaScript package manager:
+
+> Note: Since most JavaScript package managers are unable to install packages from git monorepos we provide read-only mirrors of each plugin. This makes installation option 2 more ergonomic to use.
+
+```sh
+pnpm add @tauri-apps/plugin-barcode-scanner
+# or
+npm add @tauri-apps/plugin-barcode-scanner
+# or
+yarn add @tauri-apps/plugin-barcode-scanner
+
+# alternatively with Git:
+pnpm add https://github.com/tauri-apps/tauri-plugin-barcode-scanner#v2
+# or
+npm add https://github.com/tauri-apps/tauri-plugin-barcode-scanner#v2
+# or
+yarn add https://github.com/tauri-apps/tauri-plugin-barcode-scanner#v2
+```
+
+## Usage
+
+First you need to register the core plugin with Tauri:
+
+`src-tauri/src/main.rs`
+
+```rust
+fn main() {
+    tauri::Builder::default()
+        .plugin(tauri_plugin_barcode_scanner::init())
+        .run(tauri::generate_context!())
+        .expect("error while running tauri application");
+}
+```
+
+Afterwards all the plugin's APIs are available through the JavaScript guest bindings:
+
+```javascript
+import { scan } from "@tauri-apps/plugin-barcode-scanner";
+
+// `windowed: true` actually sets the webview to transparent
+// instead of opening a separate view for the camera
+// make sure your user interface is ready to show what is underneath with a transparent element
+scan({ windowed: true, formats: [""] })
+```
+
+## Contributing
+
+PRs accepted. Please make sure to read the Contributing Guide before making a pull request.
+
+## Sponsors
+
+<table>
+  <tbody>
+    <tr>
+      <td align="center" valign="middle">
+        <a href="https://crabnebula.dev/" target="_blank">
+          <img src="sponsors/crabnebula.svg" alt="CrabNebula" width="283" height="90">
+        </a>
+      </td>
+      <td align="center" valign="middle">
+        <a href="https://impierce.com/" target="_blank">
+            <img src="sponsors/impierce.png" alt="Impierce" width="283" height="90">
+        </a>
+      </td>
+    </tr>
+  </tbody>
+</table>
```

Comment:
(cc @amrbashir @tauri-apps/tauri-board @lorenzolewis)

In the interest of not blocking useful functionality based on a sponsors policy discussion, I'm inclined to have the discussion separately. Would appreciate a gut-check from a non-CN Board Director to be sure.

But I have one immediate check I'd like to do.
@lucasfernog as PR author can you confirm that: both CrabNebula and Impierce are aware and OK with being credited?

And to be aware Tauri was pretty much caught off guard here so there's no hard commitments on whether, how and where these credits / logos might go after a followup discussion.

For further discussion I'd like to track in https://github.com/tauri-apps/governance-and-guidance since my main devils advocate question here would be: what would a different company have to do to get the same treatment? Of getting their plugin adopted by `@tauri-apps` like an official plugin, with sponsor callouts and everything. And who is responsible for future maintenance, etc.

Reply:
Thanks @Beanow , I've opened an issue in https://github.com/tauri-apps/governance-and-guidance/issues/42 to discuss this further, pending a non-CrabNebula board member to weigh in on this.

Reply:
@Beanow Impierce is aware that we are credited but wouldn't complain at all if it is removed. Our goal is not too show up in the credits, but rather sponsor this development without any particular conditions ;)

Reply:
@JelleMillenaar that is awesome! And thank you for coming in to clarify :100: